### PR TITLE
GIT UI ( squashed )

### PR DESCRIPTION
**Pull request by bunsenstraat**
Squashed version of GIT UI

### CODE REVIEW

File: apps/remix-ide-e2e/src/githttpbackend/package.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "scripts": {
+    "start:server": "npx ts-node server.ts"
+  },
+  "dependencies": {
+    "body-parser": "^1.20.2",
+    "child_process": "^1.0.2",
+    "express": "^4.19.2",
+    "git-http-backend": "^1.1.2",
+    "path": "^0.12.7",
+    "zlib": "^1.0.5"
```
Comment (LianaHus) [inline apps/remix-ide-e2e/src/githttpbackend/package.json:11]: can you add linter rules?

Comment (bunsenstraat) [reply apps/remix-ide-e2e/src/githttpbackend/package.json:11]: it's not an nx package

**Review by yann300 — APPROVED** [review]

File: libs/remix-ui/git/src/components/github/devicecode.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      headers: {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json'
+      },
+    });
+
+    // convert response to json
+    const response = await accestokenresponse.data;
+
+    if (response.error) {
```
Comment (LianaHus) [inline libs/remix-ui/git/src/components/github/devicecode.tsx:57]: do you need this?

File: libs/remix-ui/git/src/components/navigation/branchedetails.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  const fetchBranch = async () => {
+    await actions.fetch({
+      remote: null,
+      ref: branch,
+      singleBranch: true,
+      relative: true
+    })
+    //actions.fetch(null, branch.name, null, null, false, true)
```
Comment (LianaHus) [inline libs/remix-ui/git/src/components/navigation/branchedetails.tsx:55]: please rm

File: libs/remix-ui/git/src/components/panels/clone.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  );
+
+  const clone = async () => {
+    await actions.clone({
+      url: cloneUrl,
+      branch: cloneBranch,
+      depth: cloneDepth,
+      singleBranch: !cloneAllBranches
+    })
+    //await actions.clone(cloneUrl, cloneBranch, cloneDepth, !cloneAllBranches)
```
Comment (LianaHus) [inline libs/remix-ui/git/src/components/panels/clone.tsx:41]: rm?

File: package.json [hunk 5]
```diff
@@ -97,6 +98,9 @@
     "@ethereumjs/util": "9.0.3",
     "@ethereumjs/vm": "8.0.0",
     "@ethersphere/bee-js": "^3.2.0",
+    "@fortawesome/fontawesome-svg-core": "^6.5.1",
+    "@fortawesome/free-solid-svg-icons": "^6.5.1",
+    "@fortawesome/react-fontawesome": "^0.2.0",
```
Comment (LianaHus) [inline package.json:103]: can we avoid this?

**Comment by LianaHus** [comment]
i think linter is not working on your code. can you add rules to the project?

**Pull request merged**